### PR TITLE
feat(docs): add a readme for feature flags

### PR DESCRIPTION
**Pull request by philbooth**
It's not perfect but it's a start. If this lands I'll add a link to it in the content server docs and anywhere else that people suggest.

@mozilla/fxa-devs r?

### CODE REVIEW

**Comment by philbooth** [comment]
Might be best reviewed by reading the rendered view:

https://github.com/mozilla/fxa-shared/blob/pb/feature-flags-readme/feature-flags/README.md

File: feature-flags/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+that can be used to control rollout
+(or rollback).
+This approach also facilitates
+mainline development,
+where engineers merge their work directly to `master`
+in regular, smaller chunks
+and releases are cut from
+dedicated branches off `master`.
+There's a much longer discussion
+to be had around mainline devleopment,
```
Comment (shane-tomlinson) [inline feature-flags/README.md]: `devleopment` => `development`

File: feature-flags/README.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+which is out of scope for this readme.
+
+Anyway,
```
Comment (shane-tomlinson) [inline feature-flags/README.md]: The history might be better off in its own section and a concise definition of feature flagging presented here.

File: feature-flags/README.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+to return to the original question,
+feature-flagging is the mechanism
+by which we can move
+from the state of affairs in the first paragraph
+to the bright new future of the second.
+
+## How is feature-flagging implemented?
```
Comment (shane-tomlinson) [inline feature-flags/README.md]: This section doesn't have a line in the jump links above.

File: feature-flags/README.md (hunk 2 above)
Comment (philbooth) [reply feature-flags/README.md]: Fwiw, I'm less interested in objectively defining what feature-flagging is than I am in conveying the cultural context for our team, that led me to work on it.

**Comment by philbooth** [comment]
@shane-tomlinson I haven't addressed your editorial feedback because I'm not sure I follow it all / am persuaded by it. Going to merge anyway then it's easier for you to just edit it how you want, rather than me try to interpret your instructions (and likely get it wrong).

**Pull request merged**